### ISSUE TITLE
Running Boot processes from IDE client

### ISSUE THREAD

**Issue by BoykoAlex** [spring-projects/sts4#1190]
Currently to connect to running Spring Boot Apps the Boot LS process uses JVM internal APIs to get all running java processes and then user can select one of these processes.
This approach won't work with the Graal VM native image of the Boot LS. We can limit the list of java processes to those launched via the IDE client (VSCode/Eclipse)
(cc: @martinlippert )

**Comment by martinlippert** [comment]
Yes, makes sense to me. We could include a preference that switches this behavior back to "show all JVM processes on the machine", in case the language server runs on a regular JVM (not a native image). But the default behavior should be to show the list of processes launched via the IDE client.

It would also make the list of processes a lot cleaner when they show up in the menu (quick pick or drop-down) to select a process, e.g. when changing log levels, connect to the process, etc.
